File: Samples/Text/DrawText/C#/DrawText.cs
using System;
using System.Diagnostics;

namespace BitMiracle.Docotic.Pdf.Samples
{
    public static class DrawText
    {
        public static void Main()
        {
            // NOTE:
            // Without a license, the library won't allow you to create or read PDF documents.
            // To get a free time-limited license key, use the form on
            // https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE");

            string pathToFile = "DrawText.pdf";

            using (var pdf = new PdfDocument())
            {
                PdfCanvas canvas = pdf.Pages[0].Canvas;

                canvas.DrawString(10, 50, "Hello, world!");

                const string LongString = "Lorem ipsum dolor sit amet, consectetur adipisicing elit";
                var singleLineOptions = new PdfTextDrawingOptions(new PdfRectangle(10, 70, 40, 150))
                {
                    Multiline = false,
                    HorizontalAlignment = PdfTextAlign.Left,
                    VerticalAlignment = PdfVerticalAlign.Top
                };
                canvas.DrawText(LongString, singleLineOptions);

                var multiLineOptions = new PdfTextDrawingOptions(new PdfRectangle(70, 70, 40, 150))
                {
                    HorizontalAlignment = PdfTextAlign.Left,
                    VerticalAlignment = PdfVerticalAlign.Top
                };
                canvas.DrawText(LongString, multiLineOptions);

                pdf.Save(pathToFile);
            }

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}");

            Process.Start(new ProcessStartInfo(pathToFile) { UseShellExecute = true });
        }
    }
}

File: Samples/Text/DrawText/VB.NET/DrawText.vb
Imports BitMiracle.Docotic
Imports BitMiracle.Docotic.Pdf

Namespace BitMiracle.Docotic.Pdf.Samples
    Public NotInheritable Class DrawText
        Public Shared Sub Main()
            ' NOTE:
            ' Without a license, the library won't allow you to create or read PDF documents.
            ' To get a free time-limited license key, use the form on
            ' https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE")

            Dim pathToFile As String = "DrawText.pdf"

            Using pdf As New PdfDocument()
                Dim canvas As PdfCanvas = pdf.Pages(0).Canvas

                canvas.DrawString(10, 50, "Hello, world!")

                Const LongString As String = "Lorem ipsum dolor sit amet, consectetur adipisicing elit"
                Dim singleLineOptions = New PdfTextDrawingOptions(New PdfRectangle(10, 70, 40, 150)) With
                {
                    .Multiline = False,
                    .HorizontalAlignment = PdfTextAlign.Left,
                    .VerticalAlignment = PdfVerticalAlign.Top
                }
                canvas.DrawText(LongString, singleLineOptions)

                Dim multiLineOptions = New PdfTextDrawingOptions(New PdfRectangle(70, 70, 40, 150)) With
                {
                    .HorizontalAlignment = PdfTextAlign.Left,
                    .VerticalAlignment = PdfVerticalAlign.Top
                }
                canvas.DrawText(LongString, multiLineOptions)

                pdf.Save(pathToFile)
            End Using

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}")

            Process.Start(New ProcessStartInfo(pathToFile) With {.UseShellExecute = True})
        End Sub
    End Class
End Namespace
